tests/mpi/renumber_cuthill_mckee.cc

```diff
@@ -22,6 +22,7 @@
 #include "../tests.h"
 #include <deal.II/base/logstream.h>
 #include <deal.II/base/mpi.h>
+#include <deal.II/base/mpi.templates.h>
```

Comment:
These two tests use `Utilities::MPI::internal::mpi_type_id`. Since it is in the internal name space I moved it out of `mpi.h`.